test-apps/test-led-matrix/src/main.cpp

```diff
@@ -0,0 +1,71 @@
+#include "LEDMatrix.h"
+#include "hw_bridge.h"
+#include "mbed.h"
+
+int main() {
+  printf("####### STARTING LED MATRIX TEST #######\r\n");
+
+  LEDMatrix blinky(D8, D9, D10);
+
+  while (true) {
+    blinky.setState(HWBRIDGE::LEDMATRIX::LEDMatrixState::SOLID_RED);
+    ThisThread::sleep_for(5s);
+    blinky.setState(HWBRIDGE::LEDMATRIX::LEDMatrixState::SOLID_BLUE);
+    ThisThread::sleep_for(5s);
+    blinky.setState(HWBRIDGE::LEDMATRIX::LEDMatrixState::SOLID_GREEN);
+    ThisThread::sleep_for(5s);
+    blinky.setState(HWBRIDGE::LEDMATRIX::LEDMatrixState::OFF);
+    ThisThread::sleep_for(5s);
+    blinky.setState(HWBRIDGE::LEDMATRIX::LEDMatrixState::FLASHING_RED);
+    ThisThread::sleep_for(5s);
+    blinky.setState(HWBRIDGE::LEDMATRIX::LEDMatrixState::FLASHING_BLUE);
+    ThisThread::sleep_for(5s);
+    blinky.setState(HWBRIDGE::LEDMATRIX::LEDMatrixState::FLASHING_GREEN);
+    ThisThread::sleep_for(5s);
+    blinky.setState(HWBRIDGE::LEDMATRIX::LEDMatrixState::OFF);
```

Comment:
this could be put in a for loop that iterated over the HWBRIDGE::LEDMATRIX enum (assuming u just wanted to test them all and didnt care about a particular order